nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,180 @@
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.sun.source.util.Trees;
+import com.uber.nullaway.fixserialization.qual.AnnotationConfig;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import javax.lang.model.element.Element;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import org.w3c.dom.Document;
+
+/** Config class for Fix Serialization package. */
+public class FixSerializationConfig {
+
+  /**
+   * If activated, the corresponding output file will be cleared and for all reported errors,
+   * NullAway will serialize information and suggests type changes to resolve them, in case these
+   * errors could be fixed by adding a {@code @Nullable} annotation. These type change suggestions
+   * are in form of {@link com.uber.nullaway.fixserialization.out.SuggestedFixInfo} instances and
+   * will be serialized at output directory. If deactivated, no {@code SuggestedFixInfo} will be
+   * created and the output file will remain untouched.
+   */
+  public final boolean suggestEnabled;
+  /**
+   * If activated, serialized information of a fix suggest will also include the enclosing method
+   * and class of the element involved in error. Finding enclosing elements is costly and will only
+   * be computed at request.
+   */
+  public final boolean suggestEnclosing;
+  /** The directory where all files generated/read by Fix Serialization package resides. */
+  public final String outputDirectory;
+
+  public final AnnotationConfig annotationConfig;
+
+  public final Serializer serializer;
+
+  public FixSerializationConfig() {
+    suggestEnabled = false;
+    suggestEnclosing = false;
+    annotationConfig = new AnnotationConfig();
+    outputDirectory = null;
+    serializer = null;
+  }
+
+  public FixSerializationConfig(
+      boolean suggestEnabled,
+      boolean suggestEnclosing,
+      AnnotationConfig annotationConfig,
+      String outputDirectory) {
+    this.suggestEnabled = suggestEnabled;
+    this.suggestEnclosing = suggestEnclosing;
+    this.outputDirectory = outputDirectory;
+    this.annotationConfig = annotationConfig;
+    serializer = new Serializer(this);
+  }
+
+  /**
+   * Sets all flags based on their values in the configuration file.
+   *
+   * @param configFilePath Path to the serialization config file written in xml.
+   */
+  public FixSerializationConfig(String configFilePath) {
+    Preconditions.checkNotNull(configFilePath);
+    Document document;
+    try {
+      DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder builder = factory.newDocumentBuilder();
+      document = builder.parse(Files.newInputStream(Paths.get(configFilePath)));
+      document.normalize();
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Error in reading/parsing config at path: " + configFilePath + "\n" + e);
```

Comment:
Actually, and I know we don't do this consistently elsewhere in NullAway either, but... could we pass the inner exception to the `RuntimeException` constructor as a cause? To get a more complete stack trace.

Reply:
I did not fully understand what I have to do, would you please give me more details, thank you very much.

Reply:
Now I do know what I have to do :)

f5097027fad7814e22a027ccd8e8f17af16b4338